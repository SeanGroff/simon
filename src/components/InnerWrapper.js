// @flow
import React from 'react';
import styled from 'styled-components';
import Button from './Button';

const InnerWrapper = styled.div`position: relative;`;

const ButtonsRow = styled.div`margin-bottom: -4px;`;

export default (props: {}) =>
  <InnerWrapper>
    <ButtonsRow>
      <Button topLeft clickable={false} color="#00A74A" />
      <Button topRight clickable={false} color="#9F0F17" />
    </ButtonsRow>
    <ButtonsRow>
      <Button bottomLeft clickable={false} color="#CCA707" />
      <Button bottomRight clickable={false} color="#094A8F" />
    </ButtonsRow>
  </InnerWrapper>;
