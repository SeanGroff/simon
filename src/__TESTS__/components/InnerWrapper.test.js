import React from 'react';
import { shallow } from 'enzyme';
import InnerWrapper from '../../components/InnerWrapper';

it('renders correctly', () => {
  const component = shallow(<InnerWrapper />);
  expect(component).toMatchSnapshot();
});
