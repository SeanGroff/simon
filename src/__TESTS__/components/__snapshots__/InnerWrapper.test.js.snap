// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <styled.div>
    <styled.div>
        <Button
            clickable={false}
            color="#00A74A"
            topLeft={true}
        />
        <Button
            clickable={false}
            color="#9F0F17"
            topRight={true}
        />
    </styled.div>
    <styled.div>
        <Button
            bottomLeft={true}
            clickable={false}
            color="#CCA707"
        />
        <Button
            bottomRight={true}
            clickable={false}
            color="#094A8F"
        />
    </styled.div>
</styled.div>,
  "nodes": Array [
    <styled.div>
      <styled.div>
            <Button
                  clickable={false}
                  color="#00A74A"
                  topLeft={true}
            />
            <Button
                  clickable={false}
                  color="#9F0F17"
                  topRight={true}
            />
      </styled.div>
      <styled.div>
            <Button
                  bottomLeft={true}
                  clickable={false}
                  color="#CCA707"
            />
            <Button
                  bottomRight={true}
                  clickable={false}
                  color="#094A8F"
            />
      </styled.div>
</styled.div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <Unknown />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {},
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <styled.div>
          <styled.div>
                    <Button
                              clickable={false}
                              color="#00A74A"
                              topLeft={true}
                    />
                    <Button
                              clickable={false}
                              color="#9F0F17"
                              topRight={true}
                    />
          </styled.div>
          <styled.div>
                    <Button
                              bottomLeft={true}
                              clickable={false}
                              color="#CCA707"
                    />
                    <Button
                              bottomRight={true}
                              clickable={false}
                              color="#094A8F"
                    />
          </styled.div>
</styled.div>,
        "_debugID": 2,
        "_renderedOutput": <styled.div>
          <styled.div>
                    <Button
                              clickable={false}
                              color="#00A74A"
                              topLeft={true}
                    />
                    <Button
                              clickable={false}
                              color="#9F0F17"
                              topRight={true}
                    />
          </styled.div>
          <styled.div>
                    <Button
                              bottomLeft={true}
                              clickable={false}
                              color="#CCA707"
                    />
                    <Button
                              bottomRight={true}
                              clickable={false}
                              color="#094A8F"
                    />
          </styled.div>
</styled.div>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Unknown />,
}
`;
